### PR TITLE
Add some tests for user delete with submitter/candidate.

### PR DESCRIPTION
**Pull request by reiterl**
Add the Cascading delete to the two fields.

### CODE REVIEW

File: tests/system/action/user/test_delete.py [hunk 1]
```diff
@@ -58,6 +58,52 @@ def test_delete_with_speaker(self) -> None:
...
+    def test_delete_with_submitter(self) -> None:
+        self.set_models(
+            {
+                "user/111": {
+                    "username": "username_srtgb123",
+                    "submitted_motion_$_ids": ["1"],
+                    "submitted_motion_$1_ids": [34],
+                },
+                "meeting/1": {},
+                "motion_submitter/34": {"user_id": 111},
```
Comment (r-peschke) [inline tests/system/action/user/test_delete.py:98]: I would prefer to see more values for better understanding:
```
                "motion_submitter/34": {"user_id": 111, "motion_id": 50},
                "motion/50": {"submitter_ids": [34]}
            }
        )
        response = self.request("user.delete", {"id": 111})

        self.assert_status_code(response, 200)
        self.assert_model_deleted("user/111", {'submitted_motion_$1_ids': [34], 'submitted_motion_$_ids': ['1']})
        self.assert_model_deleted("motion_submitter/34", {'user_id': 111, "motion_id": 50})
        self.assert_model_exists("motion/50", {'submitter_ids': []})
```

File: tests/system/action/user/test_delete.py [hunk 2]
```diff
@@ -58,6 +58,52 @@ def test_delete_with_speaker(self) -> None:
...
+                    "title": "test_assignment",
+                    "candidate_ids": [34],
+                    "meeting_id": 1,
+                },
+            }
+        )
+        response = self.request("user.delete", {"id": 111})
+
+        self.assert_status_code(response, 200)
+        self.assert_model_deleted("user/111")
```
Comment (r-peschke) [inline tests/system/action/user/test_delete.py:85]: What do you think about these asserts:
```
        self.assert_model_deleted("user/111", {'assignment_candidate_$1_ids': [34], 'assignment_candidate_$_ids': ['1']})
        self.assert_model_deleted("assignment_candidate/34", {'assignment_id': 123, 'user_id': 111})
        self.assert_model_exists("assignment/123", {'candidate_ids': []})
```

**Review by r-peschke — CHANGES_REQUESTED** [review]
Look at my proposals for more verbose asserts.

File: tests/system/action/user/test_delete.py (hunk 1 above)
Comment (reiterl) [reply tests/system/action/user/test_delete.py]: Updated the tests.

File: tests/system/action/user/test_delete.py (hunk 2 above)
Comment (reiterl) [reply tests/system/action/user/test_delete.py]: Updated the test.

**Review by r-peschke — APPROVED** [review]

**Pull request merged**